python/ray/function_manager.py

```diff
@@ -315,6 +309,42 @@ def get_task_counter(self, job_id, function_descriptor):
             job_id = ray.JobID.nil()
         return self._num_task_executions[job_id][function_id]
 
+    def compute_collision_identifier(self, function_or_class):
+        """The identifier is used to detect excessive duplicate exports.
+
+        The identifier is used to determine when the same function or class is
+        exported many times. This can yielf false positives.
+
+        Args:
+            function_or_class: The function or class to compute an identifier
+                for.
+
+        Returns:
+            The identifier. Note that different functions or classes can give
+                rise to same identifier. However, the same function should
+                hopefully always give rise to the same identifier. TODO(rkn):
+                verify if this is actually the case. Note that if the
+                identifier is incorrect in any way, then we may give warnings
+                unnecessarily or fail to give warnings, but the application's
+                behavior won't change.
+        """
+        if sys.version_info[0] >= 3:
+            import io
+            string_file = io.StringIO()
+            if sys.version_info[1] >= 7:
+                dis.dis(function_or_class, file=string_file, depth=2)
```

Comment:
This `depth=...` keyword arg is introduced in 3.7. I imagine that it helps, but I haven't tried it out.